src/plugins/auto_batch/src/remote_context.hpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include <memory>
+
+#include "openvino/runtime/iremote_context.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/remote_context.hpp"
+
+namespace ov {
+namespace autobatch_plugin {
+
+class Plugin;
+class AutoBatchRemoteContext : public ov::RemoteContext {
+public:
+    friend class ov::autobatch_plugin::Plugin;
+    AutoBatchRemoteContext(const ov::RemoteContext& remote_context) : ov::RemoteContext(remote_context) {}
+};
```

Comment:
I suppose we should inherit IRemoteContext, and fw the call of the auto batch context to wrapped hardware context?

Reply:
Base Ilya's propose, auto batch remote context just needs to inherit from ov::RemoteContext.

Reply:
BTW, is it possible to use different remote_properties to distinguish above 2 remote contexts?
`std::shared_ptr<ov::IRemoteContext> get_default_context(const ov::AnyMap& remote_properties)`

Reply:
seems can refer to this implementation:
https://github.com/openvinotoolkit/openvino/pull/18468/files

Reply:
In the PR https://github.com/openvinotoolkit/openvino/pull/18468 I created helpers which already reused for auto plugin.
Hope it should help to have the common approach for meta plugins.
@songbell About get_context... From my point of view hw plugin should pass to the constructor of plugin specific compiled model the right context.
I think yes we can do method get_context() virtual in case if this problem cannot be solved for meta plugins